### DISCUSSION TITLE
Simplify Override Call

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#5868] [Language Ideas]
Sample C# Code with .net 6.0 / Visual Studio 2022

```
class BaseClass
{
    public virtual void ShowString()
    {
        Console.WriteLine("String 1");
    }
}

class DriveClass : BaseClass
{
    public override void ShowString()
    {
        base.ShowString();

        Console.WriteLine("String 2");
    }
}

class MainClass
{
    static void Main()
    {
        new DriveClass().ShowString();
    }
}
```

If you need to override a method, and call base method, you have to write the full name. Is it possible to simplify this step, modify it to the following simplified way? You just need to call ```base()```, and you don't need to remember the method/function name. 

```
class DriveClass : BaseClass
{
    public override void ShowString()
    {
        base();

        Console.WriteLine("String 2");
    }
}
```

This step is error-prone when your function names are similar, or when you may to change those function names later.
Because of specifying the name of the function, in some cases you have to check that the correct base class method name is called.

```
class DriveClass : BaseClass
{
    public override void ShowString()
    {
        base.ShowString();

        Console.WriteLine("String 2");
    }
}
```

**advantage**:

1. Simplifies the amount of code input and reduces the difficulty of program development.
2. It is **guaranteed** by the compiler that the **correct** base class function name is always called.
3. It saves the code review time to manually check whether the base class function is called correctly in the later stage.
4. In any override method body, it is easy to quickly distinguish whether the same name of the base class method is called, or another function name of the base class is called.

**Comment by Joe4evr** [comment]
I swear this is a duplicate, but I can't find it.

**Reply by ghost** [reply]
@Joe4evr , Do you know the result of the discussion?

**Comment by TahirAhmadov** [comment]
Heck, if we were to do this, why not:
```cs
class DriveClass : BaseClass
{
    public override void ShowString() : base()
    {
        Console.WriteLine("String 2");
    }
}
```

**Reply by Rekkonnect** [reply]
This would only mean that you call the base method before everything else within the override, and at that point it would be better to only write that in the method's body for clarity about the order of execution.

**Comment by ghost** [comment]
@TahirAhmadov  If you use my statement, you can control the printing **sequence**, the following two ways will produce different output, you can control its order internally.

```
    public override void ShowString()
    {
        base();

        Console.WriteLine("String 2");
    }
```
And:
```
    public override void ShowString()
    {
        Console.WriteLine("String 2");

        base();
    }
```

Of course your method is fine too, It is assumed that the base class method needs to be called first.

**Comment by Rekkonnect** [comment]
What would be the syntax for calling the base method when there are parameters, for example:
```csharp
public override void Something(int value)
{
     base(value);
}
```

In that case, since method arguments are still being passed intact and ordered exactly the same, the expression can even be shortened down to just `base`. This would mean:
```csharp
public override int CalculateBucket(int entry, out int index)
{
    return base + 1;
    // translates into
    return base(entry, out index) + 1;
    return base.CalculateBucket(entry, out index) + 1;
}
```
